keps/1432-admission-check-per-resource-flavor/README.md

```diff
@@ -0,0 +1,287 @@
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+Add new field to FlavorQuota API, so it contains list of AdmissionChecks thatshould run when the ResourceFlavor is used.
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+As a user who has reserved machines at my cloud provider I would like to use them first. If they are not obtainable then
+switch to spot machines, using ProvisioningRequest. It only makes sense to run Provisioning AdmissionCheck when the ResourceFlavor is Spot.
+
+
+
+### Notes/Constraints/Caveats (Optional)
+All `AdmissionChecks` assigned to a Workload must have different controllers.
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+We propose adding a new field to `FlavorQuota` API called `AdmissionChecks` in following manner:
+
+```
+type FlavorQuotas struct {
+  // name of this flavor. The name should match the .metadata.name of a
+  // ResourceFlavor. If a matching ResourceFlavor does not exist, the
+  // ClusterQueue will have an Active condition set to False.
+  Name ResourceFlavorReference `json:"name"`  
+
+  // resources is the list of quotas for this flavor per resource.
+  // There could be up to 16 resources.
+  // +listType=map
+  // +listMapKey=name
+  // +kubebuilder:validation:MinItems=1
+  // +kubebuilder:validation:MaxItems=16
+  Resources []ResourceQuota `json:"resources"`
+
+  // admissionChecks is the list of AdmissionChecks that should run when the ResourceFlavor is used.
+  // No two AdmissionChecks should have the same controller.
+  // +listType=set
+  AdmissionChecks []string `json:"admissionchecks"`
+}
+```
+
+At the same time, we want to preserve the existing `AdmissionChecks` field in `ClusterQueue` API. A Workload should have
```

Comment:
@mwielgus Apart from maintaining backwards compatibility (which I don't want to discount, though AdmissionChecks are still relatively young), is there a good reason to have AdmissionChecks at both the CQ and Flavor level?

I suppose not having to reason about AdmissionCheck override makes the API simpler to understand, and easier to debug when something is misbehaving

Reply:
Even though it's new, it's already being used in production, so I definitely would leave it supported.

Reply:
Right, but we still could consider allowing only for one level of AdmissionChecks - either CQ or RF, as suggested also in https://github.com/kubernetes-sigs/kueue/pull/1935#discussion_r1549503754

Reply:
Maybe we could simplify it and skip overriding as there are not that many use cases behind it. However, I would keep having AdmissionChecks both at the CQ and Flavor level, and I've added a Story to show an example. WDYT?

Reply:
I agree with @gabesaba and @mimowo https://github.com/kubernetes-sigs/kueue/pull/1935#discussion_r1551617863